fe1-web/model/objects/HDWallet.ts

```diff
@@ -0,0 +1,127 @@
+import { WalletStore } from 'store/stores/WalletStore';
+import { WalletCryptographyHandler } from './WalletCryptographyHandler';
+
+/**
+ * bip39 library used for seed generation and verification
+ * https://www.npmjs.com/package/bip39
+ */
+const bip39 = require('bip39');
+
+export interface WalletState {
+  seed: ArrayBuffer;
+}
+
+/**
+ * This class is responsible for all functions regarding
+ * the hierarchical deterministic wallet.
+ * The seed is created and verified here.
+ * The tokens are created and retrieved here.
+ */
+export class HDWallet {
+  private static readonly PREFIX: string = 'm';
+
+  private static readonly PURPOSE: string = '888';
+
+  private static readonly ACCOUNT: string = '0';
+
+  private static readonly SEPARATOR: string = '/';
+
+  /* cryptography manager: encrypts the seed with RSA key stored in IndexedDB */
+  private cryptoManager!: WalletCryptographyHandler;
+
+  /* local copy of encrypted seed */
+  private encryptedSeed!: ArrayBuffer;
+
+  /**
+   * this method initializes the wallet:
+   * 1. creating the crypto-manager (RSA key in IndexedDB).
+   * 2. encrypting the seed.
+   * 3. storing it in the application state through reducer.
+   * @param mnemonic the 12-word representation of the wallet seed
+   * @return true if the given mnemonic is valid, false otherwise
+   */
+  public async initialize(mnemonic: string): Promise<boolean> {
+    if (mnemonic === undefined) {
+      throw new Error('Error while initializing the wallet seed: seed is undefined');
+    }
+    const seedIsValid: boolean = bip39.validateMnemonic(mnemonic);
+    if (seedIsValid) {
+      this.cryptoManager = new WalletCryptographyHandler();
+      const seed: Uint8Array = await bip39.mnemonicToSeed(mnemonic);
+      await this.cryptoManager.initWalletStorage();
+      const encryptedSeed = await this
+        .encryptSeedToStoreInState(seed);
+      this.encryptedSeed = encryptedSeed;
+      WalletStore.store(encryptedSeed);
+    }
+    return seedIsValid;
```

Comment:
When you have a clear path which validates the inputs, it's often cleaner to immediately exit the function on a validation fail. Example:

```suggestion
    const seedIsValid: boolean = bip39.validateMnemonic(mnemonic);
    if (!seedIsValid) {
      return false;
    }
    
    this.cryptoManager = new WalletCryptographyHandler();
    const seed: Uint8Array = await bip39.mnemonicToSeed(mnemonic);
    await this.cryptoManager.initWalletStorage();
    const encryptedSeed = await this
      .encryptSeedToStoreInState(seed);
    this.encryptedSeed = encryptedSeed;
    WalletStore.store(encryptedSeed);
    return true;
```

Reply:
Yes totally cleaner